plugins/woocommerce/tests/e2e-pw/tests/merchant/products/add-variable-product/create-product-attributes.spec.js

```diff
@@ -133,7 +133,7 @@ test.describe( 'Add product attributes', { tag: '@gutenberg' }, () => {
 						'options=woocommerce_task_list_reminder_bar_hidden'
 					)
 			);
-			await page.getByRole( 'button', { name: 'Update' } ).click();
+			await page.locator( '#publish' ).click();
```

Comment:
There are two "Update" button on the Pressable website, and `name: 'Update'` returned the error: `Error: locator.click: Error: strict mode violation: getByRole('button', { name: 'Update' }) resolved to 2 elements:`. Example [here](https://a8c-woo-test-reports.s3.amazonaws.com/reports/other/on_demand_e2eexternalrun-all-e2e-tests-8e66f00/default-pressable-core-e2e/index.html#suites/41f7ab25b13debdacc7aff2751ac2043/c8c470b28ae8a62e/). Screenshot below: 

<img width="1673" alt="Screenshot 2024-08-25 at 22 54 56" src="https://github.com/user-attachments/assets/228245e6-9baa-4afc-b9ae-5e011e695ca9">

Reply:
There may be more ways to automate this by still using that `getByRole`, feels like. I have nothing against that simple locator use, so I'll approve of speeding things up. 👍

Reply:
> I'll approve of speeding things up

Let's not hurry please, and do things right instead. There is no pressure to merge this quickly.

Reply:
@Stojdza This should work with a nested locator: `page.getBySomething('locator for the sidebar').getByRole( 'button', { name: 'Update' } ).click();`

Reply:
Here is the change [f23ea2f](https://github.com/woocommerce/woocommerce/pull/50942/commits/f23ea2f27a2f07f0a25187a22c1b835861ecbaf8)

The only way I found to use `getBy*().getByRole()` was `.getByText( 'Copy to a new draft Move to' ).getByRole( 'button', { name: 'Update' } ).click();`. Playwright UI screenshot:

<img width="1254" alt="Screenshot 2024-08-27 at 13 28 36" src="https://github.com/user-attachments/assets/2b99bc30-b94c-4201-9237-287281801ede">

If any of you have some other solution, please let me know.

Reply:
That `getByText` is so weird. I would stick to the locator and already merge this. :)

Reply:
I'm fine with any approach that works and that follows the current structure. In this case, I tried both of yours proposals, and the test is more stable with `page.locator('#publishing-action').getByRole( 'button', { name: 'Update' } )`. It failed with `page.locator( '#publish :text("Update")' ).click();`